naff/client/utils/text_utils.py

```diff
@@ -0,0 +1,20 @@
+import re
+import naff.models as models
+
+__all__ = ("mentions",)
+
+
+def mentions(text: str, query: "str | re.Pattern[str] | models.BaseUser | models.BaseChannel | models.Role") -> bool:
+    """Checks whether a query is present in a text."""
+    # don't use match/case witch type(query) since subclasses aren't checked
+    if isinstance(query, str):
+        return query in text
+    elif isinstance(query, re.Pattern):
+        return query.match(text) is not None
+    elif isinstance(query, models.BaseUser):
+        # mentions with <@!ID> aren't detected without the replacement
+        return (query.mention in text.replace("@!", "@")) or (query.username in text)
```

Comment:
Why are you checking if the username is in the text? `LordOfPolls` isn't a mention `<@174918559539920897>` is

Further, if you removed that superfluous check you could squash this and the below if statements into one

Reply:
I don't want to check for (Discord) mentions, but for the presence itself (if it makes sense).

Reply:
That seems counter-intuitive to the feature's name, nor how most people would assume it works. 

Assuming we did go that route, however, using `display_name` would be better surely.

Reply:
*at least I would check for `BaseUser.tag` -> `LordOfPolls#1010`

Reply:
> That seems counter-intuitive to the feature's name, nor how most people would assume it works. 
> 
> Assuming we did go that route, however, using `display_name` would be better surely. 

Any other suggestions for the features name?

Reply:
I would suggest removing the username detection and keeping the name as it is. ;)

Reply:
Username detection really feels prone to false positives.  I definitely wouldn't want it by default.

Reply:
like so?
https://github.com/NAFTeam/NAFF/blob/da514aa1f91f422a55224fd7ba0da322b5a90df9/naff/client/utils/text_utils.py#L23-L25